tests/e2e/default/client-side-transitions/create-user-for-contacts.wdio-spec.js

```diff
@@ -0,0 +1,736 @@
+      const originalContactId = ORIGINAL_USER.contact._id;
+
+      await commonPage.goToPeople(originalContactId);
+      await submitReplaceUserForm(REPLACE_USER_FORM_TITLE);
+      // No logout triggered
+      await commonPage.goToReports();
+      const reportId = await reportsPage.getLastSubmittedReportId();
+      const basicReportId = await submitBasicForm();
+
+      // Replace user report created
+      const replaceUserReport = await utils.getDoc(reportId);
+      assertReplaceUserReport(replaceUserReport, originalContactId);
+      const { replacement_contact_id } = replaceUserReport.fields;
+      // New contact created
+      const newContact = await utils.getDoc(replacement_contact_id);
+      assert.equal(newContact.phone, ORIGINAL_USER.phone);
+      // Basic report not re-parented
+      const basicReports = await utils.getDoc(basicReportId);
+      assert.equal(basicReports.contact._id, originalContactId);
+
+      await sentinelUtils.waitForSentinel();
+      const { transitions } = await sentinelUtils.getInfoDoc(originalContactId);
+
+      // Transition not triggered
+      assert.isUndefined(transitions.create_user_for_contacts);
+      // Original contact not updated
+      const updatedOriginalContact = await utils.getDoc(DEFAULT_USER_CONTACT_DOC._id);
+      assert.isUndefined(updatedOriginalContact.user_for_contact);
+    });
+  });
+
+  it('does not create a new user when the replace_user form is submitted for online user', async () => {
+    await utils.updateSettings(SETTINGS, 'sentinel');
+    await loginAsOnlineUser();
+    const originalContactId = ONLINE_USER.contact._id;
+    await commonPage.goToPeople(originalContactId);
+
+    await submitReplaceUserForm(REPLACE_USER_FORM_TITLE);
+    // No logout triggered
+    await commonPage.goToReports();
+    const reportId = await reportsPage.getLastSubmittedReportId();
+
+    // Report is submitted successfully
+    const replaceUserReport = await utils.getDoc(reportId);
+    assertReplaceUserReport(replaceUserReport, originalContactId);
+    const { replacement_contact_id } = replaceUserReport.fields;
+
+    await sentinelUtils.waitForSentinel();
+    const { transitions } = await sentinelUtils.getInfoDoc(originalContactId);
+
+    // Transition not triggered
+    assert.isUndefined(transitions.create_user_for_contacts);
+    // Original contact not updated
+    const updatedOriginalContact = await utils.getDoc(originalContactId);
+    assert.isUndefined(updatedOriginalContact.user_for_contact);
+    // Original user not disabled
+    const [oldUserSettings] = await utils.getUserSettings({ name: ONLINE_USER.username });
+    assert.isUndefined(oldUserSettings.inactive);
+    // New user not created
+    const newUserSettings = await utils.getUserSettings({ contactId: replacement_contact_id });
+    assert.isEmpty(newUserSettings);
+    // No messages sent
+    const queuedMsgs = await getQueuedMessages();
+    assert.isEmpty(queuedMsgs);
```

Comment:
Please add an e2e test where the transition fails, for example because the user doesn't have a phone number assigned, and assert over reports created before replace failed and after replace failed. I am mostly interested in: these reports being accepted by the server and about which submitter they get (the `contact` property).